src/Akeneo/UserManagement/back/Infrastructure/Cli/Registry/AuthenticatedAsAdminCommandRegistry.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\UserManagement\Infrastructure\Cli\Registry;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class AuthenticatedAsAdminCommandRegistry
+{
+    private array $commandsAuthenticatedAsAdminUser = [];
+
+    public function registerCommand(string $commandName): void
+    {
+        $this->commandsAuthenticatedAsAdminUser[] = $commandName;
```

Comment:
```suggestion
        $this->commandsAuthenticatedAsAdminUser[$commandName] = $commandName;
```

Could be indexed in order to replace the `\in_array` below by `\array_key_exists` which is a lot quicker